### ISSUE TITLE
[SOLVED] Not working with youtube videos.

### ISSUE THREAD

**Issue by aasutossh** [woodruffw/ff2mpv#23]
As the title said it didn't work with yt, then I tried it with vimeo, it worked.
So I tried to manually play the video by passing the link to mpv, it also didn't work. 
I got errors like 
```
[ytdl_hook] youtube-dl failed: unexpected error ocurred
Failed to recognize file format.
```
```
[ytdl_hook] youtube-dl failed: unexpected error ocurred
```
Now I was sure that it's not `ff2mpv`'s issue. It warned to check if I had the latest version of `youtube-dl`. 
So I visit https://github.com/ytdl-org/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl and install it manually.
Now everything works like a charm.
Thank you @woodruffw for this gem.

tldr: Make sure your `mpv` and `youtube-dl` is latest version and enjoy.

**Comment by woodruffw** [comment]
Thanks for self-diagnosing!

**Comment by aasutossh** [comment]
Hi, is this possible to select video quality while playing youtube on mpv via ff2mpv?

**Comment by woodruffw** [comment]
> Hi, is this possible to select video quality while playing youtube on mpv via ff2mpv?

You should use your MPV configuration (`~/.config/mpv/mpv.conf` or wherever you put yours) for this.

**Comment by aasutossh** [comment]
That's doable.
Sometimes I am at a place with good internet, sometimes I am at slow one. So changing mpv settings each time would not be perfect(optimal).
While the `Play with mpv` could have (sub)context menu for this(just for YT maybe; I don't know what mpv uses to play videos from other sites).
Don't you think it's possible?
I could help maybe.

**Comment by woodruffw** [comment]
I think it’s possible, although it isn’t something that I currently want to support in ff2mpv. You’re welcome to modify your local copy, though!

**Comment by aasutossh** [comment]
Okay, Thank you for assisting.

**Comment by toprak** [comment]
> > Hi, is this possible to select video quality while playing youtube on mpv via ff2mpv?
> 
> You should use your MPV configuration (`~/.config/mpv/mpv.conf` or wherever you put yours) for this.

@woodruffw Since I do not familiar with mpv, would you mind the tell me the which setting were you mentioning there.

**Comment by woodruffw** [comment]
You can set something like this in your MPV config:

```
ytdl-format='bestvideo[width<=1920][height<=1080]+bestaudio/best'
```

That's the setting I use: it limits the video source to the best input that's at or below 1080p, and selects the best audio source regardless or codec or size.

Search for `ytdl-format` in the MPV manpages (`man mpv`) and the `FORMAT SELECTION` section in the youtube-dl manpages (`man youtube-dl`) for full documentation and examples.
